Source/GUI/Plot.h

```diff
@@ -103,14 +111,273 @@ struct compareX
+            m_items.clear();
+        }
+
+        void add(const QString& value, const QColor& color, const QString& name) {
+            m_items.append(Condition(&m_engine, m_stats, m_plotGroup));
+            m_items.back().update(value, color, name);
+        }
+
+        void add() {
+            m_items.append(Condition(&m_engine, m_stats, m_plotGroup));
+        }
+
+        void remove() {
+            m_items.removeLast();
+        }
+
+        void update(int i, const QString& conditionString, const QColor& color, const QString& name) {
+            auto & condition = m_items[i];
+            qDebug() << "updateCondition: " << i << ", string = " << conditionString << ", color = " << color << ", name = " << name;
+            condition.update(conditionString, color, name);
+        }
+
+        void updateAll(int bitdepth)
+        {
+            QList<QPair<QString, QString>> autocomplete;
+            autocomplete << QPair<QString, QString>("y", "y value of chart");
+
+            auto & engine = m_engine;
+            auto plotGroup = m_plotGroup;
+
+            engine.globalObject().setProperty("yHalf", (m_stats->y_Max[m_plotGroup] - m_stats->y_Min[m_plotGroup]) / 2);
+            autocomplete << QPair<QString, QString>("yHalf", QString("(plot max - plot min) / 2 ({%1})").arg(engine.globalObject().property("yHalf").toInt()));
+
+            auto pow2 = engine.evaluate("function(value) { return Math.pow(value, 2); }");
+            engine.globalObject().setProperty("pow2", pow2);
+            autocomplete << QPair<QString, QString>("pow2", "pow2(exponent)");
+
+            auto pow = engine.evaluate("function(base, exponent) { return Math.pow(base, exponent); }");
+            engine.globalObject().setProperty("pow", pow);
+            autocomplete << QPair<QString, QString>("pow", "pow(base, exponent)");
+
+            if(bitdepth == 0)
+            {
+                qWarning("bitdepth is 0, assuming 8...");
+                bitdepth = 8;
+            }
+
+            m_bitdepth = bitdepth;
+
+            if(plotGroup == Group_Y || plotGroup == Group_U || plotGroup == Group_V || plotGroup == Group_YDiff || plotGroup == Group_UDiff || plotGroup == Group_VDiff)
+            {
+                engine.globalObject().setProperty("maxval", ::pow(2, bitdepth));
+                autocomplete << QPair<QString, QString>("maxval", QString("2^bitdepth - %1").arg(engine.globalObject().property("maxval").toInt()));
+
+                engine.globalObject().setProperty("minval", 0);
+                autocomplete << QPair<QString, QString>("minval", QString("0"));
+
+                if(plotGroup == Group_Y || plotGroup == Group_YDiff)
+                {
+                    engine.globalObject().setProperty("broadcastmaxval", 235 * (::pow(2, bitdepth - 8)));
+                    autocomplete << QPair<QString, QString>("broadcastmaxval", QString("235 * (2^(bitdepth - 8)) - %1").arg(engine.globalObject().property("broadcastmaxval").toInt()));
```

Comment:
The `-` before the `%1` is confusing as it makes the value appear to be part of an equation such as `235 * (2^(bitdepth - 8)) - 940`. I suggest (untested) `QString("235 * (2^(bitdepth - 8)), [%1]")`

Reply:
agree

Reply:
need this same change on maxval, broadcastmaxval, and broadcastminval